game-service/src/main/java/com/lms/gameservice/controller/GameController.java

```diff
@@ -281,31 +363,59 @@ private String extractUidFromMessage(String message) {
         return null; // Handle errors
   
     }
+    
     /**
      * Used to test the Results Table is working.
      * 
      */
-    // @PostMapping("/resultTest")
-    // public void uploadResultsTest() {
+    @PostMapping("/resultTest")
+    public void uploadResultsTest() {
         
-    // LocalDate today = LocalDate.now();
+    LocalDate today = LocalDate.now();
     
-    // LocalDate lastMonday = today.minusWeeks(1).with(TemporalAdjusters.previousOrSame(DayOfWeek.MONDAY));
-    // LocalDate lastSunday = today.with(TemporalAdjusters.previous(DayOfWeek.SUNDAY));
-    
-    // DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
-    // String startDate = lastMonday.format(formatter);
-    // String endDate = lastSunday.format(formatter);
+    LocalDate lastMonday = today.minusWeeks(1).with(TemporalAdjusters.previousOrSame(DayOfWeek.MONDAY));
+    LocalDate lastSunday = today.with(TemporalAdjusters.previous(DayOfWeek.SUNDAY));
+    lastSunday = lastSunday.plusDays(1);
+
+    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    String startDate = lastMonday.format(formatter);
+    String endDate = lastSunday.format(formatter);
     
-    // List<Matches> matches = info.fetchMatchesWithinDateRange(startDate, endDate);
+    List<MatchesDTO> matches = info.fetchMatchesWithinDateRange(startDate, endDate);
 
-    // Results result = new Results();
-    // ArrayList<String> winners = new ArrayList<>();
-    // for (Matches match : matches) {
-    //     winners.add(match.getResult());
-    // }
-    // result.setWinners(winners);
+    Results result = new Results();
+    ArrayList<String> winners = new ArrayList<>();
+    for (MatchesDTO match : matches) {
+        winners.add(match.getResult());
+    }
+    result.setWinners(winners);
 
-    // resultsRepository.save(result);
-    // }
+    resultsRepository.save(result);
+    }
+
+    /**
+     * Used to test the rounds working properly.
+     * 
+     */
+    @PostMapping("/updateTest")
+    public void updateActiveGames() {
+        List<Game> games = gameRepository.findByStatus("ACTIVE");
+
+        for (Game game : games) {
+            gameService.nextRound(game.getId());
+        }
+    }
+
+    /**
+     * Used to test the rounds working properly.
+     * 
+     */
+    @PostMapping("/{game_id}/gameStartTest")
+    public void startGame(@PathVariable("game_id") int gameId) {
+        
+        Game game = gameRepository.findById(gameId)
+                .orElseThrow(() -> new IllegalArgumentException("Game not found"));
+
+                gameService.startGame(game.getId());
+    }
```

Comment:
So will an admin have to set this? Or will this be automatically ran

Reply:
no this should happen automatically from a similar function GameScheduler. I will make it so only admins can use the three test endpoints